pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -359,13 +359,13 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 		waitForControllersToBeActive = g.Add(flow.Task{
 			Name:         "Waiting until kube-controller-manager is active",
 			Fn:           flow.TaskFn(botanist.WaitForKubeControllerManagerToBeActive).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			SkipIf:       !(cleanupShootResources && kubeControllerManagerDeploymentFound),
+			SkipIf:       !cleanupShootResources || kubeControllerManagerDeploymentFound,
```

Comment:
```suggestion
		      	SkipIf:       !cleanupShootResources || !kubeControllerManagerDeploymentFound,
```

Otherwise we'd skip if cleanup is requested and KCM deployment is found.